iac-extensions/arm/src/main/java/org/sonar/iac/arm/tree/impl/bicep/InterpolatedStringImpl.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * SonarQube IaC Plugin
+ * Copyright (C) 2021-2023 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.iac.arm.tree.impl.bicep;
+
+import org.sonar.iac.arm.tree.api.bicep.InterpolatedString;
+import org.sonar.iac.arm.tree.impl.AbstractArmTreeImpl;
+import org.sonar.iac.arm.tree.impl.bicep.interpstring.InterpolatedStringLeftPiece;
+import org.sonar.iac.arm.tree.impl.bicep.interpstring.InterpolatedStringMiddlePiece;
+import org.sonar.iac.arm.tree.impl.bicep.interpstring.InterpolatedStringRightPiece;
+import org.sonar.iac.common.api.tree.Tree;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class InterpolatedStringImpl extends AbstractArmTreeImpl implements InterpolatedString {
```

Comment:
The already existing `InterpolatedStringImpl` https://github.com/SonarSource/sonar-iac/blob/98f63404bc8951f0e8ebccf165afb949177e75d3/iac-extensions/arm/src/main/java/org/sonar/iac/arm/tree/impl/bicep/InterpolatedStringImpl.java
 should be renamed to `StringCompleteImpl` according to the grammar from this ticket. (It was to simplify another ticket).

Reply:
Did it; had some problems with the fact that `StringCompleteImpl` quite reasonably implements `TextTree`, while for `InterpolatedStringImpl` it's a bit tricky (as described in [this comment](https://github.com/SonarSource/sonar-iac/pull/792#discussion_r1257923448)).